Plugin fails to load on Purpur due to missing HikariCP on UNIX OS

**Summary:**
When starting my Purpur Minecraft server, attempting to load the HandShaker plugin (version v5.0.0-1.21, hand-shaker-merged-5.0.1-1.21.jar) fails with a `NoClassDefFoundError: com/zaxxer/hikari/HikariConfig`. The stack trace points to the PlayerHistoryDatabase class attempting to use HikariCP, but the dependency is not found.

**Steps to reproduce:**
1. Download Purpur 1.21.10 server.
2. Place the HandShaker v5.0.0 (hand-shaker-merged-5.0.1-1.21.jar) in the plugins folder.
3. Start server.
4. Observe the following error in console:

```
Error occurred while enabling HandShaker v5.0.0-1.21 (Is it up to date?)
java.lang.NoClassDefFoundError: com/zaxxer/hikari/HikariConfig
    at hand-shaker-merged-5.0.1-1.21.jar/me.mklv.handshaker.paper.PlayerHistoryDatabase.initialize(PlayerHistoryDatabase.java:32) ~[hand-shaker-merged-5.0.1-1.21.jar:?]
    ...
Caused by: java.lang.ClassNotFoundException: com.zaxxer.hikari.HikariConfig
```

**Expected behavior:**
The plugin should load on a typical Purpur/Paper server without requiring users to install extra .jar dependencies, or it should clearly document any required additional libraries.

**Additional information:**
- This error appears to happen because HikariCP is not shaded/embedded.
- There is no guidance about extra setup for dependencies.
- Possible solutions:
  - Provide a “shaded”/all-in-one jar with all dependencies included, or
  - Add install docs explaining any required manual dependency setup, or
  - Add a config option to disable player history feature if not needed (skip DB init on startup), or
  - Catch this error and gracefully degrade with a warning.
- I’m running Purpur 1.21.10.

--- a/build.gradle
+++ b/build.gradle
@@ -135,10 +135,11 @@ afterEvaluate {
         group = 'build'
         description = 'Merges the Fabric and Paper jars into one combined jar'
 
-        dependsOn(tasks.named('signFabricJar')) // Ensure signing happens before merging
+        dependsOn(tasks.named('signFabricJar'))
+        dependsOn(project(':paper').tasks.named('shadowJar'))
 
         def signedJarProvider = tasks.named('signFabricJar').flatMap { it.outputs.files.elements.map { fs -> fs.first().asFile } }
-        def paperJarProvider = project(':paper').tasks.named('jar').flatMap { it.archiveFile }
+        def paperJarProvider = project(':paper').tasks.named('shadowJar').flatMap { it.archiveFile }
 
         from(signedJarProvider.map { file -> zipTree(file) })
         from(paperJarProvider.map { file -> zipTree(file) }) {

--- a/paper/build.gradle
+++ b/paper/build.gradle
@@ -1,5 +1,6 @@
 plugins {
     id 'java'
+    id 'com.github.johnrengelman.shadow' version '8.1.1'
 }
 
 group = 'me.mklv'
@@ -37,6 +38,26 @@ processResources {
     dependsOn rootProject.tasks.named('copyCert')
     filteringCharset = 'UTF-8'
     filesMatching('plugin.yml') { expand(version: project.version) }
+}
+
+shadowJar {
+    archiveClassifier.set('')
+
+    relocate 'com.zaxxer.hikari', 'me.mklv.handshaker.libs.hikari'
+    relocate 'org.sqlite', 'me.mklv.handshaker.libs.sqlite'
+    relocate 'org.slf4j', 'me.mklv.handshaker.libs.slf4j'
+
+    dependencies {
+        exclude(dependency('io.papermc.paper:paper-api'))
+    }
+}
+
+jar {
+    enabled = false
+}
+
+tasks.named('assemble') {
+    dependsOn shadowJar
 }
 
 publishing {

